packages/core/src/middleware/shift.ts

```diff
@@ -148,5 +148,5 @@ export const limitShift =
     return {
       [mainAxis]: mainAxisCoord,
       [crossAxis]: crossAxisCoord,
-    };
+    } as Coords;
```

Comment:
Is it possible to not cast here?

Reply:
If you don't cast you get:

```ts
TS2322: Type '(middlewareArguments: MiddlewareArguments) => { [x: string]: number; }' is not assignable to type '(middlewareArguments: MiddlewareArguments) => Coords'.   Type '{ [x: string]: number; }' is missing the following properties from type 'Coords': x, y
```

I'm not great at typescript but the only way I can think of not casting is:

```ts
    const coordMap = {
      [mainAxis]: mainAxisCoord,
      [crossAxis]: crossAxisCoord,
    };

    return {
      x: coordMap['x'],
      y: coordMap['y']
    }
```